### ISSUE TITLE
still no access over vlan

### ISSUE THREAD

**Issue by petettx** [gregcube/unifi-mimic#4]
"This hack is no longer needed. Direct connections now work over vlans/ subnets. Not sure when Unifi introduced this but it seems to have happened quietly."

Are you sure about that? For me it still doesn't work over vlans.

**Comment by gregcube** [comment]
I *think* so.  I'm on an Android phone, which is on a different vlan than my unifi nvr. My firewall allows these two networks to talk to each other.  The Unifi Protect app reports, "You are connected directly" -- without unifi-mimic running on my rpi4.

**Comment by petettx** [comment]
That's weird. In my case it doesn't. I even allowed all port to be accessed. The same time the unifi network app works fine and says "you are directly connected".
Which version of the app do you have?

**Comment by jirkapas** [comment]
The question is how does UBNT intend for this cross vlan access to work. I haven't yet had the time to research or test this but I suspect that it only works with cloud connected NVRs and the app just gets the NVRs local IP via UBNTs server and then blindly tries to connect to this IP every time its on a WLAN network. I don't really see any other way for this to work since there is no way for the app to discover devices behind a router.

**Comment by gregcube** [comment]
> That's weird. In my case it doesn't. I even allowed all port to be accessed. The same time the unifi network app works fine and says "you are directly connected". Which version of the app do you have?

App version: 1.15.0

I may stand corrected.  It's been awhile.  After some inspection I just realized unifi-mimic is indeed running on my rpi4.  It must have been rebooted at some point and automatically started.  I killed unifi-mimic and "You are directly connected!" no longer appears; restarted unifi-mimic and it's working again.  I could've sworn at some point it was working without unifi-mimic.  Maybe something changed (again?) on the UBNT side.  When I have some time I'm going to experiment some more.

EDIT: I have a VPN setup with wireguard on my network (another vlan) and that used to run unifi-mimic as well.  I needed to re-enable it.  I'm certain it was working before without.
